### PR TITLE
Med 191 add changes to appointments api

### PR DESCRIPTION
**Pull request by Mati-Pereira**
# Add Pagination and filter to Appointments route

#### features

 - user can see today, future and past appointments by day
 - local doctor can see his and all appointments
 
     **PR checklist:**

    Common
    - [x] types for input and output params
    - [x]  no any, should be also added to eslint rules
    - [x]  try/catch for any async function
    - [x] no magic numbers
    - [x]  compare only with constants not with strings
    (instead of user === ‘admin’, better user === roles.admin)
    - [x]  no ternary operator inside ternary operator (bad example: question ? first:
    second ? third: fourth)
    - [x]  avoid similar types with duplicating by generics
    - [x]  no console.log in pr
    - [x]  .env file should be in .gitignore
    - [x]  delete commented code if it's not part of documentation
    - [x]  no links in the code, env links should be in env file (for example: server url),
    constant links (for example default avatar URL) should be in constant file
    - [x]  constants and function names should be lowercase.
    - [x]  no dates format in the code (like "dd/MM/yyyy”), move it in global constant
    variable
    - [x]  import rules. imports should come in a specific order: node modules, absolute
    path, relative path
    - [x]  strings should be in the constant variable. Example: instead of ‘15 3 * * *’,
    const EACH_DAY_15h_15min = ‘15 3 * * *’

    1.2 Backend
    - [x]  swagger for each endpoint
    - [x]  less requests to db, all data should be taken with one query
    - [x]  public in method use only is use function externally
    - [x] use ConfigService instead of process.env

### CODE REVIEW

**Review by skazzp — APPROVED** [review]

File: src/modules/appointments/appointments.service.ts [hunk 1]
```diff
@@ -1,21 +1,33 @@
...
 import { Appointment } from '@entities/Appointments';
+import { Doctor } from '@entities/Doctor';
+
 import { CreateAppointmentDto } from '@modules/appointments/dto/create-appointment.dto';
-import { PaginationOptionsDto } from './dto/pagination-options.dto';
+import { AllPaginationCalendarOptionsDto } from '@modules/appointments/dto/allPaginationCalendar-options.dto';
+import { AllPaginationListOptionsDto } from '@modules/appointments/dto/allPaginationList-options.dto';
+import { FuturePaginationOptionsDto } from '@modules/appointments/dto/futurePagination-options.dto';
 
 @Injectable()
```
Comment (lyudad) [inline src/modules/appointments/appointments.service.ts:24]: maybe I messed something, does this routs closed?

Comment (Mati-Pereira) [reply src/modules/appointments/appointments.service.ts:24]: Sorry, i don't know what you mean. You mean close bracket?

Comment (Mati-Pereira) [reply src/modules/appointments/appointments.service.ts:24]: If it is, yes, is closed.

Comment (Mati-Pereira) [reply src/modules/appointments/appointments.service.ts:24]: If you mean protected, no. Local and Remote use this route

**Pull request merged**